Misc/NEWS.d/next/Build/2024-09-23-11-27-25.gh-issue-123990.d6HrYC.rst

```diff
@@ -0,0 +1 @@
+Remove ``WITH_FREELISTS`` macro and ``--with-freelists`` build configuration
```

Comment:
It's the ``--without-freelists`` configuration that got removed, right?

(The `--with-freelists` configuration was the default)